src/flexmeasures_client/client.py

```diff
@@ -0,0 +1,280 @@
+                    except asyncio.TimeoutError:
+                        print(
+                            f"Client request timeout occurred while connecting to the API. Retrying in {self.polling_interval} seconds..."  # noqa: E501
+                        )
+                        self.polling_step += 1
+                        await asyncio.sleep(self.polling_interval)
+                    except (ClientError, socket.gaierror) as exception:
+                        raise ConnectionError(
+                            "Error occurred while communicating with the API."
+                        ) from exception
+        except asyncio.TimeoutError as exception:
+            raise ConnectionError(
+                "Client polling timeout while connection to the API."
+            ) from exception
+
+        check_content_type(response)
+
+        return cast(dict[str, Any], await response.json()), response.status
+
+    async def request_once(
+        self,
+        method: str,
+        url: str,
+        params: dict[str, Any] | None = None,
+        headers: dict | None = None,
+        json: dict | None = None,
+    ):
+        """Sends a single request to FlexMeasures and checks the response"""
+        response = await self.session.request(
+            method=method,
+            url=url,
+            params=params,
+            headers=headers,
+            json=json,
+            ssl=self.ssl,
+        )
+        await check_response(
+            self,
+            response,
+        )
+        print(response.headers)
+        return response
+
+    def start_session(self):
+        """If there is no session, start one"""
+        if self.session is None:
+            self.session = ClientSession()
+
+    async def get_headers(self, include_auth: bool) -> dict:
+        """If the request needs to be authenticated check if there is a access_token or request one. Then create the headers dict"""  # noqa: E501
+        headers = CONTENT_TYPE_HEADERS
+        if include_auth:
+            if self.access_token is None:
+                await self.get_access_token()
+            headers |= {"Authorization": self.access_token}
+        print(headers)
+        return headers
+
+    def build_url(self, uri: str, path: str = path):
+        url = URL.build(scheme=self.scheme, host=self.host, path=path).join(
+            URL(uri),
+        )
+        return url
+
+    async def get_access_token(self):
+        """Get access token and store it on the FlexMeasuresClient."""
+        response, _status = await self.request(
+            uri="requestAuthToken",
+            path="/api/",
+            json={
+                "email": self.email,
+                "password": self.password,
+            },
+            include_auth=False,
+        )
+        print(response, _status)
+        self.access_token = response["auth_token"]
+
+    async def post_measurements(
+        self,
+        sensor_id: int,
+        start: str,
```

Comment:
I think we should accept `str` and `datetime` or even just `datetime`. `pd.Timestamp` accepts both, so in this case, I think we just need to change the type hint to `str | datetime`.